plugins/modules/junos_hostname.py

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2021 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#############################################
+#                WARNING                    #
+#############################################
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The module file for junos_hostname
+"""
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "network",
+}
+
+DOCUMENTATION = """
+---
+module: junos_hostname
+version_added: 2.9.0
+short_description: Manage SNMP server configuration on Junos devices.
+description: This module manages SNMP server configuration on devices running Junos.
+author: Rohit Thakur (@rohitthakur2590)
+requirements:
+  - ncclient (>=v0.6.4)
+  - xmltodict (>=0.12.0)
+notes:
+  - This module requires the netconf system service be enabled on the device being managed.
+  - This module works with connection C(netconf).
+  - See L(the Junos OS Platform Options,https://docs.ansible.com/ansible/latest/network/user_guide/platform_junos.html).
+  - Tested against JunOS v18.4R1
+options:
+  running_config:
+    description:
+    - This option is used only with state I(parsed).
+    - The value of this option should be the output received from the Junos device
+      by executing the command B(show system hostname).
+    - The state I(parsed) reads the configuration from C(running_config) option and
+      transforms it into Ansible structured data as per the resource module's argspec
+      and the value is then returned in the I(parsed) key within the result.
+    type: str
+  config:
+    description: A dictionary of SNMP server configuration.
+    type: dict
+    suboptions:
+      hostname:
+        description: Specify the hostname.
+        type: str
+  state:
+    description:
+    - The state the configuration should be left in.
```

Comment:
I think we should mention that the merged, replaced, and overridden states are going to behave the same for this module.